src/main/java/com/mongodb/kafka/connect/util/config/BsonTimestampParser.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.mongodb.kafka.connect.util.config;
+
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.kafka.common.config.ConfigException;
+
+import org.bson.BsonTimestamp;
+import org.bson.json.JsonReader;
+
+public final class BsonTimestampParser {
+  public static final String FORMAT_DESCRIPTION =
+      "Must be either an integer number of seconds since the Epoch in the decimal format (example: 30),"
+          + " or an instant in the ISO-8601 format (example: '1970-01-01T00:00:30Z'),"
+          + " or a BSON Timestamp in the canonical extended JSON (v2) format"
+          + " (example: '{\"$timestamp\": {\"t\": 30, \"i\": 0}}').";
+
+  public static BsonTimestamp parse(final String propertyName, final String propertyValue)
```

Comment:
Lets add a unit test for this.

Reply:
Done.

Reply:
One is an integration test, the other a unit test. Ideally, you'd only need the unit test for  testing multiple values and one integration test.

Reply:
I agree. Removed some tests from `MongoSourceConfigTest`.